CMakeLists.txt

```diff
@@ -128,8 +131,15 @@ function(ADD_PARQUET_TEST REL_TEST_NAME)
     set(TEST_PATH ${CMAKE_CURRENT_SOURCE_DIR}/${REL_TEST_NAME})
   endif()
 
-  add_test(${TEST_NAME}
-    ${BUILD_SUPPORT_DIR}/run-test.sh ${TEST_PATH})
+  if (PARQUET_TEST_MEMCHECK)
+	SET_TARGET_PROPERTIES(${TEST_NAME}
+	  PROPERTIES COMPILE_FLAGS -DPARQUET_VALGRIND)
+	add_test(${TEST_NAME}
+      valgrind --tool=memcheck --leak-check=full --error-exitcode=1 ${TEST_PATH})
```

Comment:
You could also `include(CTest)` and then use `${MEMORYCHECK_COMMAND}` which will resolve to the installation of `valgrind`.